src/lib/services/instance-stats-service.ts

```diff
@@ -171,9 +180,33 @@ export class InstanceStatsService {
             strategies,
             SAMLenabled,
             OIDCenabled,
+            clientApps,
         };
     }
 
+    async getLabeledAppCounts(): Promise<
+        { range: TimeRange; count: number }[]
+    > {
+        return [
+            {
+                range: 'allTime',
+                count: await this.clientInstanceStore.getDistinctApplicationsCount(),
+            },
+            {
+                range: '30d',
+                count: await this.clientInstanceStore.getDistinctApplicationsCount(
+                    30,
+                ),
+            },
+            {
+                range: '7d',
+                count: await this.clientInstanceStore.getDistinctApplicationsCount(
+                    7,
+                ),
+            },
+        ];
```

Comment:
This will trigger 3 queries against the DB (but right now it only runs at startup time). An alternative would be to do something like:
```sql
select t.range as [score range], count(*) as [number of occurences]
from (
  select case 
    when score between  0 and  9 then ' 0-9 '
    when score between 10 and 19 then '10-19'
    when score between 20 and 29 then '20-29'
    ...
    else '90-99' end as range
  from scores) t
group by t.range
```
which involves nested queries with temporal tables and I'm not sure which one will perform better, but I'm leaning towards keeping this simple because it's not executed frequently. 

The current query does a full scan but adding an index by last_seen might not be good because that column will be updated frequently resulting in multiple index updates.

Reply:
I agree, keep it simple. :)